theurgy.lic

```diff
@@ -1,326 +1,591 @@
+         'You drop', 'But you do not')
   end
 
-  def buy_taffelberries
-    walk_to(8280)
-    bput('order 14', 'puts some sugar-dipped taffelberries on the bar', 'You don\'t have enough money to afford that')
-    fput 'get Taffelberries'
-    put_in_container('Taffelberries')
+  def study_wall
+    get_data('spells').spell_data do |name, properties|
+      if properties['cyclic'] && DRSpells.active_spells.keys.include?(name)
+        return false
+      end
+    end
+
+    walk_to '5872'
+    move 'go stair'
+    waitfor 'Four beautifully detailed figures'
+    walk_to '5846'
+    case bput('pull candle',
+              'You tug at the silver candlestick but it',
+              'You grasp hold of the silver candlestick and pull it back')
+    when 'You grasp hold of the silver candlestick and pull it back'
+      case bput('study wall',
+                'Turning your attention to the sigils',
+                'interrupt your research')
+      when 'Turning your attention to the sigils'
+        waitfor 'as your understanding of the sigils gradually slips away.'
+      when 'interrupt your research'
+        echo 'Researching - skipping Study'
+      end
+      pause
+      move 'go hatch'
+    end
+    walk_to '5756'
+    move 'go stair'
+    waitfor 'A low relief has been carefully carved'
+    true
   end
 
-  ######################## CARVING ########################
+  def refectory
+    walk_to '5988'
+    bput('meditate', 'You bow your head and contemplate')
+    waitrt?
+    true
+  end
 
   def carve_bead
-    walk_to '1030'
-    exit unless DRRoom.npcs.empty? # Invasion check
-    return unless @immortal_aspect && exists?('carving knife') && exists?('shaper') && exists?('prayer chain')
-    prayer_bead
-  end
-
-  def prayer_bead
-    return unless inside?('holy water', @water_holder)
-    return if DRSkill.getrank('Mechanical Lore') <= 140
-    @hold_research = nil
-    fput "stow #{GameObj.right_hand.noun}" if GameObj.right_hand.name != 'Empty'
-    fput "stow #{GameObj.left_hand.noun}" if GameObj.left_hand.name != 'Empty'
-    case bput('get prayer chain', 'You get', 'I could not find', 'What were you referring to', 'already in your inventory')
-    when 'I could not find', 'What were you referring to'
+    return false unless have_holy_water?
+    walk_to @data['limb_foraging']['id']
+    stow_hands
+
+    unless get_prayer_chain?
       echo 'No Prayer Chain Found - Skipping Bead Carve'
-      return
-    when 'already in your inventory'
-      fput 'remove my prayer chain'
-      worn_chain = true
+      return false
     end
-    beads = bput('look at my chain', /^Strung on to the prayer bead chain you see .*\./, 'There are currently no beads on it.')
-    bead_count = list_to_nouns(beads.match(/^Strung on to the prayer bead chain you see (.*)\./).to_a[1])
-    if bead_count.count >= 10
-      put_in_container('prayer chain')
-      return
+
+    if count_prayer_beads >= 10
```

Comment:
You need to include support for tool belts:
  def get_item(name)
    get_crafting_item(name, @bag, @bag_items, @belt)
  end

  def stow_item(name)
    stow_crafting_item(name, @bag, @belt)
  end
...
    get_item('carving knife')
...
    stow_item('carving knife')

Also will need common-crafting in the calls and DRCC in includes.

Reply:
thats a new feature we can add after though, this scope of this is already huge

Reply:
It is simple to do and only changes a few lines and adds the two functions. Any script requiring tools must support tool belts.

Reply:
It's not supported in the original script and I can't test it since I don't have a belt.  Can you add this as a separate PR once this goes in instead, since you can test it?